server.py

```diff
@@ -15,33 +17,65 @@ def setup():
     )
     sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
     sock.bind(addr)
-    sock.listen(1)
+    sock.listen(2)
     return sock
 
 
 def response_ok():
     """Return Header and Body information for Response 200."""
-    response_headers = 'HTTP/1.1 200 OK\nContent-Type: text/html'
-    response_body = '<html><body><h1></h1>'\
-        '<p>All good here, captain.</p></body></html>'
+    response_headers = (
+        b'HTTP/1.1 200 OK\r\n'
+        + date +
+        b'Content-Type: text/html\r\n'
+        b'Content-Length:\r\n')
+
+    response_body = (
+        b'<html><body>'
+        b'<p>All good here, captain.</p>'
+        b'</body></html>')
 
     return response_headers + response_body
 
 
-def response_error():
-    """Return Header and Body information for Response 500."""
-    response_headers = 'HTTP/1.1 500 Internal Server Error\n' +\
-        'Content-Type: text/html'
-    response_body = '<html><body><h1></h1>'\
-        '<p>Your server no bueno.</p></body></html>'
+def response_error(header, text):
+    """Return Header and Body information for three types of errors"""
+
+    response_headers = (
+        header +
+        date +
+        b'Content-Type: text/html\r\n'
+        b'Content-Length:\r\n')
+
+    response_body = (
+        b'<html><body>'
+        b'<p>' + text + '</p>'
+        b'</body></html>')
 
     return response_headers + response_body
 
 
+def parse_request(request):
+    client_req = request.split('\r\n')
+    meth = client_req[0].split(' ')
+    host = ''
+    for item in client_req:
+        if "Host: " in item:
+            host = item
```

Comment:
This would be a good spot for the `else` clause of a for loop.  It would fire if Host were never found in one of the items:

```python
for item in client_req:
    if "Host" in item:
        break
else:
    raise ValueError('The required  Host header is not present')
```